enhancements/update/channel-rename-generally-available.md

```diff
@@ -0,0 +1,340 @@
+
+Beginning with 4.16, the `ga-4.y` and `fleet-approved-4.y` patterns will be initiated, making `ga-4.16` and `fleet-approved-4.16` the first in that line.
+
+### Workflow Description
+
+#### Updates into 4.16
+
+1. The cluster is running a 4.15.z release in a 4.15-capped channel like [`stable-4.15`][stable-4.15].
+1. The update service and cluster version operator [populate `status.desired.clusters`](#historical-context) with the channels compatible with the current 4.15.z, including the `*-4.15`, `candidate-4.16`, and other `*-4.16`.
+   New with this enhancement, those will include `ga-4.16` and `fleet-approved-4.16`.
+1. The cluster admin decides they want to update to 4.16.
+1. The cluster admin uses [4.16 `oc adm upgrade channel ...`][set-channel-4.16-oc] or [the 4.15 web-console][set-channel-4.15-web-console] to view the list of available channels.
+1. The cluster admin is surprised by the new `ga-4.16`, etc. and the lack of `fast-4.16`, and they click through to docs:
+  * [Web-console's need an update][web-console-channel-doc-link].
+  * `oc adm upgrade [channel]` does not include a doc link at the moment.  [Should it?](#open-questions)  If so, what kind of version skew did we want this to work for?
+1. The cluster admin selects a 4.16-capped channel like `ga-4.16`.
+1. The cluster admin initiates an update to a 4.16.z release recommended by their new channel.
+
+#### 4.16 installs
+
+1. The cluster admin decides they want to install 4.16.
+1. The cluster admin acquires a 4.16.z installer binary.
+  * Signed installer binaries are available on the mirrors, e.g. [amd64-fast-4.14][mirror-fast-4.14].
+  * [Docs][install-docs] point at [console.redhat.com][console-create] which in turn [points][console-install-user-provisioned] into the mirrors with `stable` links like [this][mirror-stable-installer].
+    This path does not currently seem to point out the existence o [the sibling GPG signature][mirror-stable-signature].
+  * Installer binaries (with hashes) are also linked from [the OCP portal page][portal-ocp], with URIs that include the version and hashes, but which do not include any channel names.
+1. The cluster admin runs the installer, which generates a ClusterVersion manifest declaring [a default channel that is manually maintained][installer-default-channel].
+   New for this enhancement, that default channel will become `fleet-approved-4.y`.
+
+#### Variation and form factor considerations
+
+HyperShift takes [`spec.channel`][HostedCluster-spec-channel] and the [`desired` Release][HostedCluster-status-version-desired] all the way up to the HostedCluster level, so the channel renames should be transparent there.
+
+Microshift manages installs via RPMs, without using an update service, so channels are not relevant there.
+
+Managed OSD/ROSA collapse channels down into channel-groups, so we'll need to work with them on that.
```

Comment:
We enforce the `stable-` channel across the fleet and curate edge availability by channel in ClusterImageSets. We're going to need to change this for all 4.16 clusters to `fleet-approved-4.y`  and ensure its in place for any cluster being installed into 4.16 and upgrading into 4.16 before we'd be able to GA it.

Sounds like this will probably break in CI for Managed OpenShift during provisioning in nightlies. cc/ @vkareh

Reply:
@maorfr for APPSRE awareness

Reply:
yes it will, however we can provide forward compatibility at the API layer by ensuring that any request carrying a `channel-group=stable` gets translates to `fleet-approved` internally. Essentially supporting both the old and new wording. This would need to be done prior to the 4.16 CIS being pushed, otherwise OCM will ignore them.

Reply:
I've added a link to [managed-upgrade operator's `inferUpgradeChannelFromChannelGroup`][1] and pointed out that OCM has a similar translation layer that it uses for hosted ROSA (which doesn't use MUO).

And yes, also the ClusterImageSet `api.openshift.com/available-upgrades` annotation, which I guess I can mention by name, even if there is no public code to link.

Anything else I'm missing that still needs capturing in the enhancement text, aside from the general exposure of "until we can test, we won't be sure we've noticed all the bits that need bumping"?

[1]: https://github.com/openshift/managed-upgrade-operator/blob/ceec30b0ad3f1d6d71cb61e6d4345db26196ee43/pkg/ocmprovider/ocmprovider.go#L202-L217

Reply:
> Do we just arbitrarily put all clusters into stable-4.Y+1?

The managed-upgrade operator [flips them into `stable-4.(y+1)` just before the update][1] (or other channel based on the target release version and [this channel-group-to-channel mapper][2], to give the CVO... well, maybe they don't wait for the CVO to form an opinion...

[1]: https://github.com/openshift/managed-upgrade-operator/blob/ceec30b0ad3f1d6d71cb61e6d4345db26196ee43/pkg/clusterversion/cv.go#L291-L307
[2]: https://github.com/openshift/managed-upgrade-operator/blob/ceec30b0ad3f1d6d71cb61e6d4345db26196ee43/pkg/ocmprovider/ocmprovider.go#L202-L218